Failed to Build for Android on Go v1.23.1 Using anet Package

Hello, Thank you for your package.

I'm encountering a build error when attempting to compile a Go project targeting Android (GOOS=android, GOARCH=arm64) using the `anet` package. The following code reproduces the issue:

```go
package main

import (
	"fmt"
	"log"
	"github.com/wlynxg/anet"
)

func main() {
	i, err := anet.Interfaces()
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(i)
}
```

**Command used:**

```bash
CGO_ENABLED=0 GOOS=android GOARCH=arm64 go build -ldflags "-s -w -checklinkname=0" main.go
```

**Error:**

```
# command-line-arguments
./main.go:10:17: undefined: anet.Interfaces
```

It appears the `anet.Interfaces` function is not available during the build process. This issue might be related to how the package handles platform-specific code.

**Environment:**

- Go version: 1.23.1
- Target platform: Android (ARM64)
- Package: `anet` (latest version)

Any insights or solutions would be appreciated.

--- a/android_api_level.go
+++ b/android_api_level.go
@@ -1,0 +1,7 @@
+//go:build !(android && cgo)
+
+package anet
+
+func androidDeviceApiLevel() int {
+	return -1
+}

--- a/android_api_level_cgo.go
+++ b/android_api_level_cgo.go
@@ -1,0 +1,23 @@
+//go:build android && cgo
+
+package anet
+
+// #include <android/api-level.h>
+import "C"
+
+import "sync"
+
+var (
+	apiLevel int
+	once     sync.Once
+)
+
+// Returns the API level of the device we're actually running on, or -1 on failure.
+// The returned value is equivalent to the Java Build.VERSION.SDK_INT API.
+func androidDeviceApiLevel() int {
+	once.Do(func() {
+		apiLevel = int(C.android_get_device_api_level())
+	})
+
+	return apiLevel
+}

--- a/interface.go
+++ b/interface.go
@@ -27,5 +27,4 @@ func InterfaceAddrsByInterface(ifi *net.Interface) ([]net.Addr, error) {
 	return ifi.Addrs()
 }
 
-// Deprecated: Android version is detected automatically.
 func SetAndroidVersion(version uint) {}

--- a/interface_android.go
+++ b/interface_android.go
@@ -1,8 +1,5 @@
 package anet
 
-// #include <android/api-level.h>
-import "C"
-
 import (
 	"bytes"
 	"errors"
@@ -15,10 +12,11 @@ import (
 )
 
 const (
-	android11ApiLevel = 30
+	android11 = 11
 )
 
 var (
+	androidVersion              uint
 	errInvalidInterface         = errors.New("invalid network interface")
 	errInvalidInterfaceIndex    = errors.New("invalid network interface index")
 	errInvalidInterfaceName     = errors.New("invalid network interface name")
@@ -30,7 +28,7 @@ type ifReq [40]byte
 
 // Interfaces returns a list of the system's network interfaces.
 func Interfaces() ([]net.Interface, error) {
-	if androidApiLevel() < android11ApiLevel {
+	if androidVersion < android11 {
 		return net.Interfaces()
 	}
 
@@ -51,7 +49,7 @@ func Interfaces() ([]net.Interface, error) {
 // The returned list does not identify the associated interface; use
 // Interfaces and Interface.Addrs for more detail.
 func InterfaceAddrs() ([]net.Addr, error) {
-	if androidApiLevel() < android11ApiLevel {
+	if androidVersion < android11 {
 		return net.InterfaceAddrs()
 	}
 
@@ -68,7 +66,7 @@ func InterfaceAddrs() ([]net.Addr, error) {
 // sharing the logical data link; for more precision use
 // InterfaceByName.
 func InterfaceByIndex(index int) (*net.Interface, error) {
-	if androidApiLevel() < android11ApiLevel {
+	if androidVersion < android11 {
 		return net.InterfaceByIndex(index)
 	}
 
@@ -114,7 +112,7 @@ func InterfaceAddrsByInterface(ifi *net.Interface) ([]net.Addr, error) {
 		return nil, &net.OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: errInvalidInterface}
 	}
 
-	if androidApiLevel() < android11ApiLevel {
+	if androidVersion < android11 {
 		return ifi.Addrs()
 	}
 
@@ -123,6 +121,13 @@ func InterfaceAddrsByInterface(ifi *net.Interface) ([]net.Addr, error) {
 		err = &net.OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: err}
 	}
 	return ifat, err
+}
+
+// SetAndroidVersion set the Android environment in which the program runs.
+// The Android system version number can be obtained through
+// `android.os.Build.VERSION.RELEASE` of the Android framework.
+func SetAndroidVersion(version uint) {
+	androidVersion = version
 }
 
 // An ipv6ZoneCache represents a cache holding partial network
@@ -417,18 +422,3 @@ func nameToFlags(name string) (net.Flags, error) {
 
 	return linkFlags(*(*uint32)(unsafe.Pointer(&ifr[syscall.IFNAMSIZ]))), nil
 }
-
-// Returns the API level of the device we're actually running on, or -1 on failure.
-// The returned value is equivalent to the Java Build.VERSION.SDK_INT API.
-var androidApiLevel = func() func() int {
-	var apiLevel int
-	var once sync.Once
-
-	return func() int {
-		once.Do(func() {
-			apiLevel = int(C.android_get_device_api_level())
-		})
-
-		return apiLevel
-	}
-}()

--- a/interface_android.go
+++ b/interface_android.go
@@ -12,11 +12,11 @@ import (
 )
 
 const (
-	android11 = 11
+	android11ApiLevel = 30
 )
 
 var (
-	androidVersion              uint
+	customAndroidApiLevel       = -1
 	errInvalidInterface         = errors.New("invalid network interface")
 	errInvalidInterfaceIndex    = errors.New("invalid network interface index")
 	errInvalidInterfaceName     = errors.New("invalid network interface name")
@@ -28,7 +28,7 @@ type ifReq [40]byte
 
 // Interfaces returns a list of the system's network interfaces.
 func Interfaces() ([]net.Interface, error) {
-	if androidVersion < android11 {
+	if androidApiLevel() < android11ApiLevel {
 		return net.Interfaces()
 	}
 
@@ -49,7 +49,7 @@ func Interfaces() ([]net.Interface, error) {
 // The returned list does not identify the associated interface; use
 // Interfaces and Interface.Addrs for more detail.
 func InterfaceAddrs() ([]net.Addr, error) {
-	if androidVersion < android11 {
+	if androidApiLevel() < android11ApiLevel {
 		return net.InterfaceAddrs()
 	}
 
@@ -66,7 +66,7 @@ func InterfaceAddrs() ([]net.Addr, error) {
 // sharing the logical data link; for more precision use
 // InterfaceByName.
 func InterfaceByIndex(index int) (*net.Interface, error) {
-	if androidVersion < android11 {
+	if androidApiLevel() < android11ApiLevel {
 		return net.InterfaceByIndex(index)
 	}
 
@@ -112,7 +112,7 @@ func InterfaceAddrsByInterface(ifi *net.Interface) ([]net.Addr, error) {
 		return nil, &net.OpError{Op: "route", Net: "ip+net", Source: nil, Addr: nil, Err: errInvalidInterface}
 	}
 
-	if androidVersion < android11 {
+	if androidApiLevel() < android11ApiLevel {
 		return ifi.Addrs()
 	}
 
@@ -126,8 +126,26 @@ func InterfaceAddrsByInterface(ifi *net.Interface) ([]net.Addr, error) {
 // SetAndroidVersion set the Android environment in which the program runs.
 // The Android system version number can be obtained through
 // `android.os.Build.VERSION.RELEASE` of the Android framework.
+// If version is 0 the actual version will be detected automatically if possible.
 func SetAndroidVersion(version uint) {
-	androidVersion = version
+	switch {
+	case version == 0:
+		customAndroidApiLevel = -1
+	case version >= 11:
+		customAndroidApiLevel = android11ApiLevel
+	default:
+		customAndroidApiLevel = 0
+	}
+}
+
+func androidApiLevel() int {
+	if customAndroidApiLevel != -1 {
+		// user-provided api level should be used
+		return customAndroidApiLevel
+	}
+
+	// try to autodetect api level
+	return androidDeviceApiLevel()
 }
 
 // An ipv6ZoneCache represents a cache holding partial network

